dashboard/test/lib/api/v1/school_autocomplete_test.rb

```diff
@@ -1,6 +1,26 @@
 require 'test_helper'
 
 class Api::V1::SchoolAutocompleteTest < ActiveSupport::TestCase
+  test 'is zip search with four digits' do
+    assert_equal true, Api::V1::SchoolAutocomplete.instance.zip_search?('1234')
+  end
+
+  test 'is zip search with extended partial zip and nothing after the hyphen' do
+    assert_equal true, Api::V1::SchoolAutocomplete.instance.zip_search?('12345-')
+  end
+
+  test 'is zip search with extended partial zip and two digits the hyphen' do
+    assert_equal true, Api::V1::SchoolAutocomplete.instance.zip_search?('12345-12')
+  end
+
+  test 'is zip search with alphabetic string' do
+    assert_equal false, Api::V1::SchoolAutocomplete.instance.zip_search?('abc')
+  end
+
+  test 'is zip search with invalid extended partial zip' do
+    assert_equal false, Api::V1::SchoolAutocomplete.instance.zip_search?('12345-ab')
+  end
```

Comment:
Great tests! 

Small nit: rename the last 2 so it's clear the expected result is false, e.g. `is zip search with invalid extended partial zip returns false`